### PR TITLE
test for jest conversion

### PR DESCRIPTION
**Pull request by jywarren**
Working from these for triggering the event without jQuery: 

- https://javascript.info/dispatch-events
- https://stackoverflow.com/a/46815122/1116657

Sourcing syntax from this: https://github.com/publiclab/PublicLab.Editor/blob/main/test/ui-testing/bold.test.js

Trying to convert this Jasmine test:

https://github.com/publiclab/PublicLab.Editor/blob/f8e1edb6ff992a03aa593feae53783f2b9867bd2/spec/javascripts/button_spec.js#L13-L21

### CODE REVIEW

File: test/ui-testing/publishButton.test.js [hunk 1]
```diff
@@ -7,8 +7,16 @@ beforeAll(async () => {
 
 describe('Publish button', () => {
   
-  test('something we are testing described here', () => {
-
+  test('Publish button is enabled', () => {
+    // Check initially that Publish button is disabled.    
+    expect(await page.evaluate(() => document.querySelector('.ple-textarea').disabled).toBe(true);
```
Comment (jywarren) [inline test/ui-testing/publishButton.test.js:12]: What's odd is that I thought we use the /Bootstrap class/ `disabled` to set the disabled state on this. I don't know how that interacts with the HTML attribute -- the difference is:

`<i class="disabled"></i>` (Bootstrap CSS)
`<i disabled></i>` (HTML)

So we may need to disambiguate how this is working, possibly by manually testing the jquery `$('.ple-publish').prop('disabled')` vs. just `el.disabled`.

**Comment by jywarren** [comment]
OK - so i think that's right:

```

  ● Publish button › Publish button is enabled

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: false

      10 |   test('Publish button is enabled', async () => {
      11 |     // Check initially that Publish button is disabled.    
    > 12 |     expect(await page.evaluate(() => document.querySelector('.ple-textarea').disabled)).toBe(true);
         |                                                                                         ^
      13 |     // Add title.
      14 |     await page.evaluate(() => {
      15 |       document.querySelector('.ple-module-title input').value = 'A title'
```

We should try this: `document.querySelector('.ple-publish').style.includes('disabled'));`

oh wait. First we should be checking `ple-publish` and not `ple-textarea`! Doh!

**Comment by jywarren** [comment]
That worked! @NARUDESIGNS 🎉 

So, maybe what we should do is to look closely at the old test vs. this new one. And to come up with a little bit of helpful documentation (even just a short checklist) for how to do this kind of test conversion successfully. It looks like a combination of:

- `.evaluate()`
- adding `async`
- switching to non-jQuery functions

Does that sound right?

**Comment by jywarren** [comment]
Would you like to try that for an additional test file now? I'll merge this. I hope this'll get us moving now! Thanks Paul, for your patience!!!

**Comment by jywarren** [comment]
Ah, and we should add to the checklist - removing the tests from the Jasmine files. So when we finish all conversions, we'll have no more Jasmine tests!

**Comment by NARUDESIGNS** [comment]
Wow awesome! 
I took some time to rest my head after our meeting. I can't wait to see those tests turn green!!!
I'd have a look at it tomorrow, it's 11pm here right now.
Thank you @jywarren you are amazing! 🔥 🔥 🔥

File: test/ui-testing/publishButton.test.js [hunk 2]
```diff
@@ -9,14 +9,14 @@ describe('Publish button', () => {
...
+    expect(await page.evaluate(() => document.querySelector('.ple-publish').disabled)).toBe(true);
     // Add title.
     await page.evaluate(() => {
       document.querySelector('.ple-module-title input').value = 'A title';
       document.querySelector('.ple-module-title input').dispatchEvent(new KeyboardEvent('keydown', { "code": "9" })); // random key
     });
     // Check final state of Publish button.
-    expect(await page.evaluate(() => document.querySelector('.ple-textarea').disabled)).toBe(false);
+    expect(await page.evaluate(() => document.querySelector('.ple-publish').disabled)).toBe(false);
   });
```
Comment (NARUDESIGNS) [inline test/ui-testing/publishButton.test.js:20]: Hi @jywarren I noticed you didn't add the `timeout` variable here. Is this part of the new change or is it something you missed out?

Comment (jywarren) [reply test/ui-testing/publishButton.test.js:20]: I think I was able to set it globally on the first few lines and so didn't need to set it per-test.

Comment (jywarren) [reply test/ui-testing/publishButton.test.js:20]: Ah, interesting, actually on checking, I am not setting the timeout. I think that means we could remove that line, actually. 

For reference, I'm stuck on a related issue (going past the timeout limit in this case) in this test https://github.com/publiclab/image-sequencer/pull/2026

https://github.com/publiclab/image-sequencer/blob/58eb3ef6b1caba88aaf217d93ec085a14765d8aa/test/ui-2/test/OverlayInput.test.js#L1-L12

That actually is globally setting timeout. But here I think i just removed it and for the purposes of our test it wasn't needed. I think the default timeout is 5 or 10 seconds? So i guess that's enough for this.

**Pull request merged**